File: TumCampusApp/src/de/tum/in/tumcampusapp/models/Tuition.java
package de.tum.in.tumcampusapp.models;

import java.util.List;

import org.simpleframework.xml.Element;
import org.simpleframework.xml.ElementList;
import org.simpleframework.xml.Root;

/**
 * Class holding tuition information.
 * <p>
 * Note: This model is based on the TUMOnline web service response format for a
 * corresponding request.
 * 
 * @author NTK
 */
@Root(name = "row", strict = false)
public class Tuition {

	@Element(name = "soll")
	private String soll;

	@Element(name = "frist")
	private String frist;

	@Element(name = "semester_bezeichnung")
	private String semester_bezeichnung;

	public String getSoll() {
		return soll;
	}

	public String getFrist() {
		return frist;
	}
	
	public String getSemesterBez() {
		return semester_bezeichnung;
	}

}

File: app/src/main/java/de/tum/in/tumcampusapp/component/tumui/tutionfees/model/Tuition.kt
package de.tum.`in`.tumcampusapp.component.tumui.tutionfees.model

import android.content.Context
import android.content.Intent
import com.tickaroo.tikxml.annotation.PropertyElement
import com.tickaroo.tikxml.annotation.Xml
import de.tum.`in`.tumcampusapp.R
import de.tum.`in`.tumcampusapp.api.tumonline.converters.DateTimeConverter
import de.tum.`in`.tumcampusapp.api.tumonline.converters.FloatConverter
import de.tum.`in`.tumcampusapp.component.tumui.tutionfees.TuitionFeesActivity
import org.joda.time.DateTime
import java.text.ParseException
import java.util.*

/**
 * Class holding tuition information.
 *
 *
 * Note: This model is based on the TUMOnline web service response format for a
 * corresponding request.
 */
@Xml(name = "row")
data class Tuition(@PropertyElement(name = "frist", converter = DateTimeConverter::class)
                   val deadline: DateTime,
                   @PropertyElement(name = "semester_bezeichnung")
                   val semester: String,
                   @PropertyElement(name = "soll", converter = FloatConverter::class)
                   val amount: Float) {

    fun getIntent(context: Context): Intent = Intent(context, TuitionFeesActivity::class.java)

    val isPaid: Boolean
        get() = amount == 0f

    fun getAmountText(context: Context): String {
        return try {
            val amountText = String.format(Locale.getDefault(), "%.2f", amount)
            return "$amountText €"
        } catch (e: ParseException) {
            context.getString(R.string.not_available)
        }
    }

}
